fix panic in linuxNodeHandler.NodeDelete

cyclinder opened the pull request:
Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

<!-- Description of change -->

Fixes: https://github.com/cilium/cilium/issues/25543
```release-note
Fix crash of cilium-agent happening when a remote node without node IP addresses is removed.
```

bimmlerd commented on pkg/datapath/linux/node_ids.go:
```diff
@@ -126,6 +126,10 @@ func (n *linuxNodeHandler) allocateIDForNode(node *nodeTypes.Node) uint16 {
+		return
+	}
+
 	nodeID := n.nodeIDsByIPs[oldNode.IPAddresses[0].IP.String()]
```
Wouldn't the following be a cleaner fix? In symmetry with `allocateIDForNode` above,  `getNodeIDForNode` has a fallback ID of zero if there are no IPs.

```suggestion
	nodeID := n.getNodeIDForNode(oldNode)
```

cyclinder replied:
sorry, I didn't notice it, now updated.

pchaigno replied:
That function was introduced just yesterday so that might be why you didn't notice when preparing the patch :smile:

cyclinder committed:
fix panic in linuxNodeHandler.NodeDelete

Observed a panic: runtime.boundsError{x:0, y:0, signed:true, code:0x0} (runtime error: index out of range [0] with length 0)

github.com/cilium/cilium/pkg/datapath/linux.(*linuxNodeHandler).deallocateIDForNode(0xc003e871f0?, 0xc003e87108?)
	/go/src/github.com/cilium/cilium/pkg/datapath/linux/node_ids.go:110 +0x335

Signed-off-by: qifeng guo <qifeng.guo@daocloud.io>
